src/components/Hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react";
+
+const useLocalStorage = <T>(key: string, initialValue: T): [T, (newValue: T) => void] => {
+    const [value, setValue] = useState<T>(initialValue);
+
+    useEffect(() => {
+        const value = window.localStorage.getItem(key);
```

Comment:
don't shadow nonlocal variables! name it something other than `value` - it's very easy to accidently be using the wrong variable like this!
maybe 'existingValue'? or you can come up with a better name :)